source/basic_skills.rst

```diff
@@ -1304,32 +1304,595 @@ If you managed to solve this, congratulations!
+(entered as ``\a``, ``\b``, and ``\g``)
+for arbitrary types.
+The library often uses letters like ``R`` and ``G``
+for the carries of algebraic structures likes rings and groups,
+respectively,
+but in general greek letters are used for types,
+especially when there is little or no structure
+associated with them.
+
+Associated to any partial order, ``≤``,
+this is also a *strict partial order*, ``<``,
+which acts somewhat like ``<`` on the real numbers.
+Saying the ``x`` is less than ``y`` in this order
+is equivalent to saying that it is less-than-or-equal to ``y``
+and not equal to ``y``.
+
+.. code-block:: lean
+
+    import order.basic
+
+    variables {α : Type*} [partial_order α]
+    variables x y z : α
+
+    -- BEGIN
+    #check x < y
+    #check (lt_irrefl x : ¬ x < x)
+    #check (lt_trans : x < y → y < z → x < z)
+    #check (lt_of_le_of_lt : x ≤ y → y < z → x < z)
+    #check (lt_of_lt_of_le : x < y → y ≤ z → x < z)
+
+    example : x < y ↔ x ≤ y ∧ x ≠ y :=
+    lt_iff_le_and_ne
+    -- END
+
+.. TODO: add reference to logic chapter
+
+In this example, the symbol ``∧`` stands for "and,"
+the symbol ``¬`` stands for "not," and
+``x ≠ y`` abbreviates ``¬ (x = y)``.
+In a later chapter, you will learn how to use
+these logical connectives to *prove* that ``<``
+has the properties indicated.
+
+A *lattice* is a structure that extends a partial
+order with operations ``⊓`` and ``⊔`` that are
+analogous to ``min`` and ``max`` on the real numbers:
+
+.. code-block:: lean
+
+    import order.lattice
+
+    variables {α : Type*} [lattice α]
+    variables x y z : α
+
+    #check x ⊓ y
+    #check (inf_le_left : x ⊓ y ≤ x)
+    #check (inf_le_right : x ⊓ y ≤ y)
+    #check (le_inf : z ≤ x → z ≤ y → z ≤ x ⊓ y)
+
+    #check x ⊔ y
+    #check (le_sup_left : x ≤ x ⊔ y)
+    #check (le_sup_right: y ≤ x ⊔ y)
+    #check (sup_le : x ≤ z → y ≤ z → x ⊔ y ≤ z)
+
+The characterizations of ``⊓`` and ``⊔`` justify calling them
+the *greatest lower bound* and *least upper bound*, respectively.
+You can type them in VS code using ``\glb`` and ``\lub``.
+The symbols are also often called then *infimum* and
+the *supremum*,
+and mathlib refers to them as ``inf`` and ``sup`` in
+theorem names.
+To further complicate matters,
+they are also often called *meet* and *join*.
+Therefore, if you work with lattices,
+you have to keep the following dictionary in mind:
+
+* ``⊓`` is the *greatest lower bound*, *infimum*, or *meet*.
+
+* ``⊔`` is the *least upper bound*, *supremum*, or *join*.
+
+Some instances of lattices include:
+
+* ``min`` and ``max`` on any total order, such as the integers or real numbers with ``≤``
+
+* ``∩`` and ``∪`` on the collection of subsets of some domain, with the ordering ``⊆``
+
+* ``∧`` and ``∨`` on boolean truth values, with ordering ``x ≤ y`` if either ``x`` is false or ``y`` is true
+
+* ``gcd`` and ``lcm`` on the natural numbers (or positive natural numbers), with the divisibility ordering, ``∣``
```

Comment:
I would add examples from algebra and topology, for psychological preparation. Maybe linear subspaces of a given vector space (with intersection and sum) and topologies on a given type (inf is the topology generated by open subsets of both sides, sup contains subsets that are open for both, because we use slightly weird order relation on topologies...).

Reply:
good idea -- I'll do that too.